generator/main.cpp

```diff
@@ -67,77 +68,157 @@ void displayHelp(GeneratorSet *generatorSet);
 
 namespace
 {
+    static const QStringList candidatesQtModules {
+      QStringLiteral("QtCore"),
+      QStringLiteral("QtGui"),
+      QStringLiteral("QtNetwork"),
+      QStringLiteral("QtOpenGL"),
+      QStringLiteral("QtXml"),
+    };
+
+    static bool isDir(const QString& p) { return QFileInfo(p).isDir(); }
+    static QString joinPath(const QString& base, const QString& child) {
+      return QDir(base).filePath(child);
+    }
 
     QStringList getIncludeDirectories(const QString &commandLineIncludes)
     {
       QStringList includes;
-      includes << QString(".");
-
-      QChar pathSplitter = QDir::listSeparator();
+      includes << QStringLiteral(".");
 
-      // Environment PYTHONQT_INCLUDE
-      QString includePath = getenv("PYTHONQT_INCLUDE");
-      if (!includePath.isEmpty())
-        includes += includePath.split(pathSplitter, Qt::SkipEmptyParts);
+      const QChar pathSplitter = QDir::listSeparator();
 
-      // Includes from the command line
-      if (!commandLineIncludes.isEmpty())
-        includes += commandLineIncludes.split(pathSplitter, Qt::SkipEmptyParts);
-      for (auto it = includes.begin(); it != includes.end();)
-      {
-        if (!QDir(*it).exists())
-        {
-          qWarning() << "Include path " << it->toUtf8() << " does not exist, ignoring it.";
-          it = includes.erase(it);
+      // From env var PYTHONQT_INCLUDE
+      const QString envInclude = qEnvironmentVariable("PYTHONQT_INCLUDE");
```

Comment:
`qEnvironmentVariable` has been introduced in Qt 5.10, so the tests using 5.9 now fail.